sys/kern/src/arch/rv64/trap.rs

```diff
@@ -11,14 +11,149 @@ use crate::startup::with_task_table;
 use crate::task;
 use abi::{FaultInfo, FaultSource};
 
-extern crate riscv_rt;
+//use riscv::register;
+#[cfg(feature = "riscv-supervisor-mode")]
+use riscv::register::{
+    scause as xcause, scause::Exception as xcauseException,
+    scause::Interrupt as xcauseInterrupt,
+    scause::Interrupt::SupervisorTimer as xInterruptTimer,
+    scause::Trap as xcauseTrap, sepc as xepc, sscratch as xscratch,
+    stval as xtval, stvec as xtvec, stvec::TrapMode as xTrapMode,
+};
```

Comment:
Why do you rename `Trap`, `Exception`, `Interrupt`, etc?

Reply:
I wanted there to be some level of consistency in the consuming functions, given I had to rename xInterruptTimer to map to the different types. I considered it undesirable to have paths and renamed types intermingled.